src/neighbors.cc

```diff
@@ -1113,8 +1112,8 @@ neighborsUdpAck(const cache_key * key, icp_common_t * header, const Ip::Address
             neighborIgnoreNonPeer(from, opcode);
         } else if (p->stats.pings_acked > 100) {
             if (100 * p->icp.counts[ICP_DENIED] / p->stats.pings_acked > 95) {
-                debugs(15, DBG_CRITICAL, "95%% of replies from '" << p->host << "' are UDP_DENIED");
-                debugs(15, DBG_CRITICAL, "Disabling '" << p->host << "', please check your configuration.");
+                debugs(15, DBG_CRITICAL, "95% of replies from " << *p << " are UDP_DENIED");
```

Comment:
> We could also improve logs by combining the two debugs() into one line.

I combined and polished these lines further in commit f9547bf.

> The if-statement can be improved a bit too. No need to waste CPU multiplying by 100, just check against 0.95.

There is such a need if we want to continue save CPU cycles by using fast integer math rather than switching to doubles. However, that is not the reason I left that (problematic for other reasons!) statement alone. I simply do not want to imply that this PR changes that (possibly buggy) condition logic.